File: frontend/src/types/chronicle.ts
export interface ChronicleEvent {
  id: number
  year: number
  date: string | null
  title: string
  content: string
  image_url: string | null
  sort_order: number
  created_at: string
}

export interface ChronicleEventPayload {
  year: number
  date?: string | null
  title: string
  content: string
  image_url?: string | null
  sort_order?: number
}

File: backend/app/api/v1/chronicle.py
from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.models.chronicle_event import ChronicleEvent
from app.schemas.chronicle import ChronicleEventRead


router = APIRouter()


@router.get("", response_model=list[ChronicleEventRead])
def list_chronicle_events(
    db: Session = Depends(get_db),
):
    statement = select(ChronicleEvent).order_by(
        ChronicleEvent.year.asc(),
        ChronicleEvent.date.asc(),
        ChronicleEvent.sort_order.asc(),
        ChronicleEvent.id.asc(),
    )

    return db.scalars(statement).all()
